src/test/groovy/com/jmeter/websocket/plugin/WebsocketUpgradeListenerSpock.groovy

```diff
@@ -0,0 +1,91 @@
+package com.jmeter.websocket.plugin
+
+import com.jmeter.websocket.plugin.endpoint.jetty.JettyWebsocketUpgradeListener
+import org.apache.jmeter.samplers.SampleResult
+import org.eclipse.jetty.websocket.api.UpgradeRequest
+import org.eclipse.jetty.websocket.api.UpgradeResponse
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+@Unroll
+class WebsocketUpgradeListenerSpock extends Specification {
+
+    @Subject
+    JettyWebsocketUpgradeListener listener = new JettyWebsocketUpgradeListener(new SampleResult())
+
+    def "Should throw NullPointerException when result is null"() {
+        when:
+        listener = new JettyWebsocketUpgradeListener(null)
+        then:
+        thrown(NullPointerException)
+    }
+
+    def "Should extract #headers upgrade request headers to sample result request headers as '#result'"() {
+        given:
```

Comment:
I would recommend to use additional tabs for inner element under label (e.g. given...)
I mean something like this
````
given:
    UpgradeRequest upgradeRequest = new UpgradeRequest('wss://127.0.0.1/websocket')
when:
    listener.onHandshakeRequest(upgradeRequest)
````
From my point of view it is more readable.